pkg/apis/internal.admin.acorn.io/v1/resources.go

```diff
@@ -60,87 +65,85 @@ func (current *Resources) Remove(incoming Resources, all bool) {
+			exceededResources = append(exceededResources, resource)
+		}
 	}
-	if current.CPU.Cmp(incoming.CPU) < 0 {
-		exceededResources = append(exceededResources, "Cpu")
+
+	// Define function for checking quantity resources to keep code DRY
+	checkQuantityResource := func(resource string, currentVal, incomingVal resource.Quantity) {
+		if currentVal.Cmp(incomingVal) <= 0 {
+			exceededResources = append(exceededResources, resource)
+		}
 	}
 
+	checkResource("Apps", current.Apps, incoming.Apps)
+	checkResource("Containers", current.Containers, incoming.Containers)
+	checkResource("Jobs", current.Jobs, incoming.Jobs)
+	checkResource("Volumes", current.Volumes, incoming.Volumes)
+	checkResource("Secrets", current.Secrets, incoming.Secrets)
+	checkResource("Images", current.Images, incoming.Images)
+	checkResource("Projects", current.Projects, incoming.Projects)
+
+	checkQuantityResource("VolumeStorage", current.VolumeStorage, incoming.VolumeStorage)
+	checkQuantityResource("Memory", current.Memory, incoming.Memory)
+	checkQuantityResource("Cpu", current.CPU, incoming.CPU)
+
 	// Build an aggregated error message for the exceeded resources
 	if len(exceededResources) > 0 {
-		return fmt.Errorf("exceeded quota for resources: %s", strings.Join(exceededResources, ", "))
+		return fmt.Errorf("quota would be exceeded for resources: %s", strings.Join(exceededResources, ", "))
 	}
 
 	return nil
 }
 
 // ToString will return a string representation of the Resources struct.
-func (current *Resources) ToString() string {
-	resources := []string{}
-	if current.Apps > 0 {
-		resources = append(resources, fmt.Sprintf("Apps:%v", current.Apps))
-	}
-	if current.Containers > 0 {
-		resources = append(resources, fmt.Sprintf("Containers:%v", current.Containers))
-	}
-	if current.Jobs > 0 {
-		resources = append(resources, fmt.Sprintf("Jobs:%v", current.Jobs))
-	}
-	if current.Volumes > 0 {
-		resources = append(resources, fmt.Sprintf("Volumes:%v", current.Volumes))
-	}
-	if current.Secrets > 0 {
-		resources = append(resources, fmt.Sprintf("Secrets:%v", current.Secrets))
-	}
-	if current.Images > 0 {
-		resources = append(resources, fmt.Sprintf("Images:%v", current.Images))
-	}
+func (current *Resources) NonEmptyString() string {
+	var resources []string
 
-	if !current.VolumeStorage.IsZero() {
-		resources = append(resources, fmt.Sprintf("VolumeStorage:%v", current.VolumeStorage))
-	}
-	if !current.Memory.IsZero() {
-		resources = append(resources, fmt.Sprintf("Memory:%v", current.Memory))
+	// Define function for checking int resources to keep code DRY
+	checkResource := func(resource string, value int) {
+		if value > 0 {
+			resources = append(resources, resource)
+		}
 	}
-	if !current.CPU.IsZero() {
-		resources = append(resources, fmt.Sprintf("CPU:%v", current.CPU))
+
+	// Define function for checking quantity resources to keep code DRY
+	checkQuantityResource := func(resource string, currentVal resource.Quantity) {
+		if currentVal.IsZero() {
```

Comment:
I think this is supposed to be inverted:

```suggestion
		if !currentVal.IsZero() {
```

Reply:
Yep!